lib/haskell/natural4/src/LS/XPile/MathLang/GenericMathLang/TranslateL4.hs

```diff
@@ -903,9 +945,25 @@ parseExpr x@(MTT str) = do
     Left _error -> trace [i|can't parse with pExpr: #{x}|] pure $ mteToLitExp x
 parseExpr x = pure $ mteToLitExp x
 
-splitGenitives :: [MTExpr] -> (Maybe MTExpr, [MTExpr])
-splitGenitives [] = (Nothing, [])
-splitGenitives (mte:mtes) = (isGenitive mte, mtes)
+-- | Given a '[MTExpr]' that looks like:
+--
+-- @[MTT "parent's", MTT "sibling's", MTT "income"]@
+--
+-- strip away the suffix @"'s"@ from the first element of the list
+-- if there is any.
+--
+-- >>> splitGenitives [MTT "parent's", MTT "sibling's", MTT "income"]
+-- Just (MTT "parent",MTT "sibling's" :| [MTT "income"])
+--
+-- >>> splitGenitives [MTI 0, MTT "parent's", MTT "income"]
+-- Nothing
```

Comment:
`cabal-doctests` dislikes this test. It will work locally if I move it after the definition of `splitGenitives`. Am I doing something obviously wrong?

Reply:
Hmm, I think it was me who wrote that function, but I'm unsure if I actually had a working test or if that was just written as documentation purposes. So whatever works for you, probably works for the others!

Reply:
I refactored this function and added the doctests to make sure I am not overlooking anything. And it looks like other doctests look the same and work, but mine doesn't :(